### DISCUSSION TITLE
Retemplating a button control

### DISCUSSION THREAD

**Discussion by npolyak** [AvaloniaUI/Avalonia#6137] [Q&A] [answered]
In WPF I like creating controls from scratch. In Avalonia, I started  by creating a button template:

    <Button HorizontalAlignment="Center"
            VerticalAlignment="Center"
            x:Name="MyButton">
      <Button.Template>
        <ControlTemplate TargetType="Button">
          <Grid Background="Red"
                Width="150"
                Height="30"/>
        </ControlTemplate>
      </Button.Template>
    </Button>

But when I click this button, it still reacts to the click. How do I disable the button default reaction in order to use e.g. Triggers instead.

**Comment by maxkatz6** [comment]
There is Transitions set which is not part of the template: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L78-L89

Something like this should override value from build-in styles:
```xaml
  <Style Selector="Button:pressed, RepeatButton:pressed, ToggleButton:pressed">
    <Setter Property="RenderTransform" Value="none" />    
  </Style>
```

**Reply by npolyak** [reply]
Thanks, I'll try to investigate it in the Avalonia code.

**Comment by maxkatz6** [comment, marked as answer]
> In WPF I like creating controls from scratch

It might make sense to not include whole FluentTheme (or the Default theme), but include your own controls' styles and only specific styles from build-in theme that you don't want to redefine (see list [here](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/FluentControls.xaml)).

With that approach you will avoid overhead from build-in theme. But obviously it will be harder to maintain, since you need to reimplement a lot of styles. Though it depends on how much you like to create controls from scratch.

**Reply by maxkatz6** [reply]
Ultimately you can avoid Fluent/Default themes at all and reimplement every style/template. Some applications based on Avalonia actually do that, when their styles are heavily customized. CC @MarchingCube

**Reply by npolyak** [reply]
Thank Max!

**Reply by npolyak** [reply]
Max, continuing the discussion - how do I disconnect the Fluent theme - seems like it was there by default - I did do anything to actually use it. How do I make the application not to use any theme by default?

**Reply by maxkatz6** [reply]
In App.xaml there is a FlyoutTheme style include. Just remove that line.

**Reply by npolyak** [reply]
cool, thx

**Reply by maxkatz6** [reply]
Note that some essential controls like Window and UserControl also has its own small styles.

**Reply by npolyak** [reply]
I think we can mark it as answered, I do not think I am allowed to do it though or at least i do not know how

**Reply by MarchingCube** [reply]
Also another 5 cents from me - if you are going to provide new custom style for most of the controls it also becomes better performance wise to also drop builtin styles and just pick whatever is needed. Otherwise you will end up with generaly 2x the styles which has a noticeable performance impact.

**Reply by npolyak** [reply]
Thanks!
